go/vt/topotools/vschema_ddl.go

```diff
@@ -28,55 +29,51 @@ import (
 
 // ApplyVSchemaDDL applies the given DDL statement to the vschema
 // keyspace definition and returns the modified keyspace object.
-func ApplyVSchemaDDL(ksName string, ks *vschemapb.Keyspace, alterVschema *sqlparser.AlterVschema) (*vschemapb.Keyspace, error) {
-	if ks == nil {
-		ks = new(vschemapb.Keyspace)
+func ApplyVSchemaDDL(ksName string, ksvs *topo.KeyspaceVSchemaInfo, alterVschema *sqlparser.AlterVschema) (*topo.KeyspaceVSchemaInfo, error) {
+	if ksvs.Tables == nil {
+		ksvs.Tables = map[string]*vschemapb.Table{}
 	}
```

Comment:
I believe the `GetVSchema` should be done inside this method to ensure all the callers of `ApplyVSchemaDDL` have a common code. Post that they can have a action method of their own which they can pass in.
This will avoid the issue of missing this call outside of `ApplyVSchemaDDL`

Reply:
Good idea. I did that here https://github.com/vitessio/vitess/pull/17401/commits/60e9de3493216bf75c2d04ea2824e19f8bb55c20